File: src/infrastructure/servers/factories/abstracts/i_server_factory.py
from abc import ABC, abstractmethod

class IServerFactory(ABC):

    @abstractmethod
    def build_server(self):
        pass

    @abstractmethod
    def require_env(self, key: str, error_message: str) -> str:
        pass

    @abstractmethod
    def optional_env(self, key: str, default: str) -> str:
        pass

    @abstractmethod
    def optional_int_env(self, key: str, default: int) -> int:
        pass

    @abstractmethod
    def optional_float_env(self, key: str, default: float) -> float:
        pass

File: angular-app/src/core/infrastructure/servers/factories/abstracts/i-server-factory.ts
import type { IServer } from '../../abstracts/i-server';

export interface IServerFactory {
  buildServer(): IServer;

  createServer(): IServer;

  requireEnv(key: string, errorMessage: string): string;

  optionalEnv(key: string, defaultValue: string): string;

  optionalIntEnv(key: string, defaultValue: number): number;

  optionalFloatEnv(key: string, defaultValue: number): number;
}
